app/views/models/index.html.erb

```diff
@@ -22,12 +22,12 @@
         <td class="ellipsis"><%= model.hardware_model %></td>
         <td class='ra'><%= link_to hosts_count[model], hosts_path(:search => "model = \"#{model}\"") %></td>
         <td class="col-md-1">
-          <%= action_buttons(display_delete_if_authorized(hash_for_model_path(:id => model).
-            merge(:auth_object => model, :authorizer => authorizer),
-            :confirm => _("Delete %s?") % model.name)) %>
+          <%= action_buttons(display_delete_action_if_authorized(:name => model.name, :id => model.id, :controller => "models", :auth_object => model, :auth_action => :destroy)) %>
         </td>
       </tr>
     <% end %>
   </tbody>
 </table>
 <%= will_paginate_with_info @models %>
+<div id="delete-modal"></div>
```

Comment:
any chance we can let the helper do this? assuming we would like to inject it to many/all pages?

Reply:
Sure we can.

Reply:
I wonder, does this work with more than one component? shouldn't each line should have its own unique div?